src/smartpeak/source/core/Parameters.cpp

```diff
@@ -173,11 +173,72 @@ namespace SmartPeak
     }
   }
 
-  void Parameter::setValueFromString(const std::string& value_as_string)
+  void Parameter::setValueFromString(const std::string& value_as_string, bool allow_change_type)
   {
     CastValue c;
     Utilities::parseString(value_as_string, c);
-    value_ = c;
+    if (allow_change_type || c.getTag() == value_.getTag())
+    {
+      value_ = c;
+    }
+    else
+    {
+      // try some cast
+      if ((c.getTag() == CastValue::Type::INT) && (value_.getTag() == CastValue::Type::FLOAT))
+      {
+        value_ = static_cast<float>(c.i_);
+      }
+      else if ((c.getTag() != CastValue::Type::STRING) && (value_.getTag() == CastValue::Type::STRING))
+      {
+        value_.s_ = value_as_string;
+      }
+    }
+  }
+
+  void Parameter::setRestrictionsFromString(const std::string& restriction_as_string)
+  {
+    if (restriction_as_string.empty())
+      return;
+    if ((restriction_as_string[0] == '[') && (restriction_as_string[restriction_as_string.size() - 1] == ']'))
```

Comment:
Is there a reason we cannot use the utilities function for converting from a string list here?